pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -173,32 +221,63 @@ func (m *mergeQuerier) mergeDistinctStringSlice(f stringSliceFunc) ([]string, st
 func (m *mergeQuerier) Close() error {
 	errs := tsdb_errors.NewMulti()
 	for pos, tenantID := range m.tenantIDs {
-		errs.Add(errors.Wrapf(m.queriers[pos].Close(), "failed to close querier for tenant id %s", tenantID))
+		errs.Add(errors.Wrapf(m.queriers[pos].Close(), `failed to close querier for {%s="%s"}`, defaultTenantLabel, tenantID))
 	}
 	return errs.Err()
 }
 
+type selectJob struct {
+	seriesSet *storage.SeriesSet
```

Comment:
This is a pointer to interface. Why do we need to do that?

Reply:
I felt like it was more readable than passing an index around, but I have changed that to use the index:
89d04df